### PR TITLE
Fix secret creation when using self provisioned elasticsearch instances

### PR DESCRIPTION
**Pull request by kevinearls**
Signed-off-by: Kevin Earls <kearls@redhat.com>

### CODE REVIEW

File: pkg/controller/jaeger/jaeger_controller_test.go [hunk 1]
```diff
@@ -4,6 +4,9 @@ import (
 	"context"
 	"testing"
 
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
```
Comment (jpkrohling) [inline pkg/controller/jaeger/jaeger_controller_test.go:8]: Import order is strange.

File: test/e2e/multiple_instances_test.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+	"context"
+	"testing"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type MulitpleInstanceTestSuite struct {
```
Comment (jpkrohling) [inline test/e2e/multiple_instances_test.go:15]:
```suggestion
type MultipleInstanceTestSuite struct {
```

File: test/e2e/multiple_instances_test.go [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+	// In production we'd use 3 nodes but 1 is sufficient for this test.
+	jaegerInstance := getJaegerSelfProvSimpleProd(jaegerInstanceName, namespace, 1)
+	createEsSelfProvDeployment(jaegerInstance, jaegerInstanceName, namespace)
+	defer undeployJaegerInstance(jaegerInstance)
+
+	// Create a second instance with the same name but in a different namespace
+	secondContext, err := createNewTestContext()
+	defer secondContext.Cleanup()
+	secondNamespace := secondContext.GetID()
+	secondjaegerInstance := getJaegerSelfProvSimpleProd(jaegerInstanceName, secondNamespace, 1)
```
Comment (jpkrohling) [inline test/e2e/multiple_instances_test.go:71]:
```suggestion
	secondJaegerInstance := getJaegerSelfProvSimpleProd(jaegerInstanceName, secondNamespace, 1)
```

File: test/e2e/multiple_instances_test.go [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+	require.NotEqual(t, string(loggingEsCrtOne), string(loggingEsCrtTwo))
+}
+
+func createNewTestContext() (*framework.Context, error) {
+	secondContext, err := prepare(t)
+	if err != nil {
+		if secondContext != nil {
+			secondContext.Cleanup()
+		}
+		require.FailNow(t, "Failed in prepare with: "+err.Error())
```
Comment (jpkrohling) [inline test/e2e/multiple_instances_test.go:95]:
```suggestion
		require.FailNow(t, "Failed in prepare with: %v", err)
```

File: test/e2e/multiple_instances_test.go [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+	}
+	require.NoError(t, err, "Failed trying to create a new test context")
```
Comment (jpkrohling) [inline test/e2e/multiple_instances_test.go:97]: If you move this require to the second line in the test, you don't need the `if err != nil` check.

File: test/e2e/utils.go [hunk 6]
```diff
@@ -729,3 +731,74 @@ func waitForElasticSearch() {
...
+				Security: v1.IngressSecurityNoneExplicit,
+			},
+			Strategy: v1.DeploymentStrategyProduction,
+			Storage: v1.JaegerStorageSpec{
+				Type: v1.JaegerESStorage,
+				Elasticsearch: v1.ElasticsearchSpec{
+					NodeCount: nodeCount,
+					Resources: &corev1.ResourceRequirements{
+						Limits:   corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("1Gi")},
+						Requests: corev1.ResourceList{corev1.ResourceMemory: resource.MustParse("1Gi")},
```
Comment (jpkrohling) [inline test/e2e/utils.go:758]: Not blocking, but could we have different values for `Requests` and `Limits`? Tests are often used as reference to check how we expect things to exist/run.

File: test/e2e/utils.go [hunk 7]
```diff
@@ -729,3 +731,74 @@ func waitForElasticSearch() {
...
+	if specifyOtelImages {
+		logrus.Infof("Using OTEL collector for %s", instanceName)
+		exampleJaeger.Spec.Collector.Image = otelCollectorImage
+		exampleJaeger.Spec.Collector.Config = v1.NewFreeForm(getOtelConfigForHealthCheckPort("14269"))
+	}
+
+	return exampleJaeger
+}
+
+func createEsSelfProvDeployment(jaegerInstance *v1.Jaeger, jaegerInstanceName, jaegerNamespace string) {
```
Comment (jpkrohling) [inline test/e2e/utils.go:774]:
```suggestion
func createESSelfProvDeployment(jaegerInstance *v1.Jaeger, jaegerInstanceName, jaegerNamespace string) {
```

File: test/e2e/multiple_instances_test.go [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+
+func (suite *MulitpleInstanceTestSuite) AfterTest(suiteName, testName string) {
+	handleTestFailure()
+}
+
+/*
+ * This test verifies that we create the elasticsearch secrets correctly if someone creates production Jaeger
+ * instances with the same name in different namespaces
+ */
+func (suite *MulitpleInstanceTestSuite) TestVerifySecrets() {
```
Comment (jpkrohling) [inline test/e2e/multiple_instances_test.go:56]: Stupid question, but are we sure this test fails when the fix isn't applied?

Comment (kevinearls) [reply test/e2e/multiple_instances_test.go:56]: @jpkrohling Yes, I ran it on OCP 4.5 with the 1.17.6 Jaeger Operator and it failed.

File: test/e2e/utils.go (hunk 6 above)
Comment (kevinearls) [reply test/e2e/utils.go:758]: @jpkrohling what would you suggest setting them to?

File: pkg/controller/jaeger/jaeger_controller_test.go (hunk 1 above)
Comment (kevinearls) [reply pkg/controller/jaeger/jaeger_controller_test.go:8]: @jpkrohling isn't make fmt supposed to fix this?

Comment (jpkrohling) [reply pkg/controller/jaeger/jaeger_controller_test.go:8]: It's complicated :-)

File: test/e2e/utils.go (hunk 6 above)
Comment (jpkrohling) [reply test/e2e/utils.go:758]: Either the limit to 2Gi, or the requests to 512Mi (which might not be enough?)

**Review by jpkrohling — APPROVED** [review]
LGTM

**Pull request merged**